lib/jekyll/admin/server/static_file.rb

```diff
@@ -40,7 +43,21 @@ def static_files
         end
 
         def static_file
-          static_files.find { |f| f.path == static_file_path }
+          file = static_files.find { |f| f.path == static_file_path }
+          if !file
+            file_list = static_files.select do |f|
+              # Files that are in this directory
+              # Joined with / to ensure user can't do partial paths
+              f.path.start_with? File.join(static_file_path, "/")
+            end.map(&:to_liquid)
+            if file_list == []
+              nil
```

Comment:
@benbalter I think I added that because `find` returns `nil` if no results are found.